watchtower-plugin/src/main.rs

```diff
@@ -126,22 +128,33 @@ async fn register(
     Ok(json!(receipt))
 }
 
-/// Gets the latest registration receipt from the client to a given tower (if it exists).
-///
+/// Gets the registration receipt(s) from the client to a given tower (if it exists) in the given
+/// range. If no range is given, then gets the latest registration receipt.
 /// This is pulled from the database
 async fn get_registration_receipt(
     plugin: Plugin<Arc<Mutex<WTClient>>>,
     v: serde_json::Value,
 ) -> Result<serde_json::Value, Error> {
-    let tower_id = TowerId::try_from(v).map_err(|x| anyhow!(x))?;
+    let params = GetRegistrationReceiptParams::try_from(v).map_err(|x| anyhow!(x))?;
+    let tower_id = params.tower_id;
+    let subscription_start = params.subscription_start;
+    let subscription_expiry = params.subscription_expiry;
     let state = plugin.state().lock().unwrap();
 
-    if let Some(response) = state.get_registration_receipt(tower_id) {
-        Ok(json!(response))
+    let response =
+        state.get_registration_receipt(tower_id, subscription_start, subscription_expiry);
+    if response.clone().unwrap().is_empty() {
+        if state.towers.contains_key(&tower_id) {
+            Err(anyhow!(
+                "No registration receipt found for {tower_id} on the given range"
+            ))
+        } else {
+            Err(anyhow!(
+                "Cannot find {tower_id} within the known towers. Have you registered?"
+            ))
+        }
```

Comment:
I don't really grasp this part. why is there an `.unwrap()`?!
`response` is an optional vector, so `Option::None` & an empty vector signal different meanings? right?

I guess `Option::None` is when the tower isn't found and an empty vector is when the registration range doesn't contain any receipts?
If so, then the `.clone().unwrap()` part will panic with no useful information to the user.

Otherwise, we don't need the `None` variant anymore.

Reply:
Yeah, that is actually what i think I was suggesting for, however, checking `DBM::load_registration_receipt` I don't think we ever produce a `None` return.

Reply:
I'd be good to test this out by creating a test that queries a tower that is not part of the DB and see whether the method is returning an empty vec or None